transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java

```diff
@@ -67,23 +67,32 @@ public View getView(@NonNull final ViewGroup viewGroup) {
                 @Override
                 public void onFocusChange(View v, boolean hasFocus) {
                     if (!hasFocus) {
-                        mValue = ((EditText) v).getText().toString();
+                        String input = ((EditText) v).getText().toString();
+                        mValue = formatToApi(input);
                         mListener.valueChanged();
                     } else {
                         mListener.widgetFocused(NumberWidget.this.getName());
                     }
                 }
             });
             editText.addTextChangedListener(new TextWatcher() {
+                boolean ignoreChange = false;
+
                 @Override
                 public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                 }
 
                 @Override
                 public void onTextChanged(CharSequence s, int start, int before, int count) {
-                    if (before != count) {
-                        mValue = s.toString();
+                    if (!ignoreChange && before != count) {
+                        mValue = formatToApi(s.toString());
                         mListener.saveTextChanged(getName(), getValue());
+
+                        ignoreChange = true;
```

Comment:
We cannot set the text in afterTextChanged, it will result in an infinite loop of events.

I tracked the number of times the if-statement got triggered for pattern `1##A@@`, I entered 4 characters: 23bc (so that would result in 123Abc
- without ignoreChange boolean: 6 executions
- with ignoreChange boolean: 4 executions

From a functionality point of view, it seems exactly the same. It's sort like doing a preventDefault